### DISCUSSION TITLE
Shopify Gift Card Support

### DISCUSSION THREAD

**Discussion by austriker27** [statamic-rad-pack/shopify#101] [Help]
I added a shopify gift card to my shopify dashboard but its showing as out of stock on the statamic side. Is there anything special I need to do to handle it? 

I noticed the `options` array has `denominations` in it: 

example md file: 

```md
---
id: 32fbd9f2-1bbf-4533-b1bc-71e9f79a110d
blueprint: product
product_id: 6945840070832
published_at: '2021-11-02 05:16:58'
title: 'Rainier Watch Shop Gift Card'
vendor:
  - rainier-watch
type:
  - gift-cards
tags:
  - gift-card
options:
  option1: Denominations
featured_image: shopify/rainier-watch__digital-gift-card.jpg
gallery:
  - shopify/rainier-watch__digital-gift-card.jpg
collections:
  - non-sale-items
  - popular-products
published: true
product_category:
  - gift-cards
---
```

i didnt see any prior discussions, issues, or anything in the docs. Thanks!

**Comment by jackabox** [comment]
Heya, Yep - these kind of bend the rules of normal products in the way they are stored. I believe, they carry no stock limits on them. It's on my radar to add special rules for these but I can't promise the time for delivery at the moment.

**Reply by austriker27** [reply]
Gotcha gotcha - thanks for the reply! Yeah I figured they were kinda weird or something. 

Full transparency - I was going to use shopify gift cards as a stop gap until this gift card app has headless support: https://govalo.com/

So maybe they will get headless support finished soon and I won't need the shopify gift cards! Im not entirely sure how their gift cards will work and if they will be different or not in how they operate on the backend

**Comment by meisbellajack** [comment]
Shopify Gift Cards definitely 'bend the rules.' Because they are digital assets, Shopify sets their inventory_policy to deny but usually leaves the quantity null or effectively infinite, which causes many integration layers (like Statamic) to default them to 'Out of Stock' because they don't see a positive integer.

A quick workaround until the 'special rules' are added:
**Manual Inventory Toggle:** In your Shopify Admin, go to the Gift Card product and check if 'Track Quantity' is enabled. If it is, manually set a high number (like 9999). Often, integrations just need to see any number > 0 to flip the status to 'In Stock.'
**The 'Denominations' Array:** Since you noticed the options array carries the denominations, you might need to ensure your Statamic template is pulling the price from the variant level rather than the top-level product. Gift cards are essentially a collection of variants with no physical 'Master' version.
**The Architecture Perspective:**
This is similar to the '3-Option Wall' we see with physical products. Shopify's native logic is very rigid. If you're building out complex gift options (like adding a custom message or a specific digital delivery date), you might eventually run into the same limit.
For our high-end builds, we've started moving toward App Embeds like [Misk Smart Product Options](https://apps.shopify.com/misk-smart-product-options?) to handle these extra 'meta' fields. It keeps the core product (or Gift Card) data clean while letting you layer on the 'Smart Logic' that integrations like Statamic sometimes struggle to parse natively.

Hope that helps bridge the gap for now!
